kin-sdk/src/androidTest/kotlin/kin/sdk/KinAccountIntegrationTest.kt

```diff
@@ -67,64 +70,57 @@ class KinAccountIntegrationTest {
-        expectedEx.expectMessage(kinAccount.publicAddress.orEmpty())
-        kinAccount.balanceSync
-    }
-
     @Test
     @LargeTest
     @Throws(Exception::class)
-    fun getStatusSync_AccountNotActivated_StatusNotActivated() {
+    fun getBalanceSync_FundedAccount_GotBalance() {
         val kinAccount = kinClient.addAccount()
-        fakeKinIssuer.createAccount(kinAccount.publicAddress.orEmpty())
 
-        val status = kinAccount.statusSync
-        assertThat(status, equalTo(AccountStatus.NOT_ACTIVATED))
-    }
+        val latch = CountDownLatch(1)
 
-    @Test
-    @LargeTest
-    @Throws(Exception::class)
-    fun getBalanceSync_FundedAccount_GotBalance() {
-        val kinAccount = kinClient.addAccount()
+        var listenerRegistration : ListenerRegistration? = null
+        listenerRegistration = kinAccount.addAccountCreationListener {
+            listenerRegistration?.remove()
+            assertThat(kinAccount.balanceSync.value(), equalTo(BigDecimal("0.00000")))
+            fakeKinIssuer.fundWithKin(kinAccount.publicAddress.orEmpty(), "3.14159")
+            assertThat(kinAccount.balanceSync.value(), equalTo(BigDecimal("3.14159")))
+            latch.countDown()
+        }
         fakeKinIssuer.createAccount(kinAccount.publicAddress.orEmpty())
 
-        kinAccount.activateSync()
-        assertThat(kinAccount.balanceSync.value(), equalTo(BigDecimal("0.0000000")))
+        assertTrue(latch.await(timeoutDurationSeconds, TimeUnit.SECONDS))
 
-        fakeKinIssuer.fundWithKin(kinAccount.publicAddress.orEmpty(), "3.1415926")
-        assertThat(kinAccount.balanceSync.value(), equalTo(BigDecimal("3.1415926")))
     }
 
     @Test
     @LargeTest
     @Throws(Exception::class)
-    fun getStatusSync_CreateAndActivateAccount_StatusActivated() {
+    fun getStatusSync_CreateAccount_StatusActivated() {
         val kinAccount = kinClient.addAccount()
+
+        val latch = CountDownLatch(1)
+
+        var listenerRegistration : ListenerRegistration? = null
+        listenerRegistration = kinAccount.addAccountCreationListener {
+            listenerRegistration?.remove()
+            assertThat(kinAccount.balanceSync.value(), equalTo(BigDecimal("0.00000")))
+            val status = kinAccount.statusSync
+            assertThat(status, equalTo(AccountStatus.ACTIVATED))
+            latch.countDown()
+        }
         fakeKinIssuer.createAccount(kinAccount.publicAddress.orEmpty())
 
-        kinAccount.activateSync()
-        assertThat(kinAccount.balanceSync.value(), equalTo(BigDecimal("0.0000000")))
-        val status = kinAccount.statusSync
-        assertThat(status, equalTo(AccountStatus.ACTIVATED))
+        assertTrue(latch.await(timeoutDurationSeconds, TimeUnit.SECONDS))
     }
 
     @Test
     @LargeTest
     @Throws(Exception::class)
-    fun activateSync_AccountNotCreated_AccountNotFoundException() {
+    fun accountNotCreated_AccountNotFoundException() {
```

Comment:
does this test pass? I don't see a reason for `AccountNotFoundException` to be thrown.

Reply:
no, of course it doesn't pass.
I should have remove the whole test.